master/internal/api_config_policies.go

```diff
@@ -74,21 +82,70 @@ func (*apiServer) GetGlobalConfigPolicies(
 }
 
 // Delete workspace task config policies.
-func (*apiServer) DeleteWorkspaceConfigPolicies(
+func (a *apiServer) DeleteWorkspaceConfigPolicies(
 	ctx context.Context, req *apiv1.DeleteWorkspaceConfigPoliciesRequest,
 ) (*apiv1.DeleteWorkspaceConfigPoliciesResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = workspace.AuthZProvider.Get().CanGetWorkspaceID(
+		ctx, *curUser, req.WorkspaceId,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var w model.Workspace
+	err = db.Bun().NewSelect().Model(&w).Where("id = ?", req.WorkspaceId).Scan(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
```

Comment:
task config policies get deleted with a workspace delete cascade, so if a workspace gets deleted, the TCPs for that workspace do as well!

Reply:
But what if something goes wrong? What if the cascade takes a long time (?) and the user interrupts it? What if someone accidentally removes the cascade and it doesn't happen anymore?

My thinking is, an attempt to delete a config policy for a workspace that doesn't exist should be a warning, not an error. I prefer not to box in the software if I don't have to; it can easily result in unintended consequences. Since all config policies should be deleted with the workspace, is it actually an error to submit the request?

I definitely tend to use an SRE-style mindset ... perhaps I'm overly cautious about strange things that could go wrong. What are your thoughts @salonig23?

Reply:
We talked about this elsewhere, but we decided to settle on requests failing if they try to delete a config policy for an invalid scope. This is to keep consistent with other API calls in our system (experiment, workspace, etc.. requests fail when trying to delete a non-existent experiment, wksp, ...)